extensions/amp-apester-media/0.1/monetization/companion/video.js

```diff
@@ -66,6 +88,50 @@ function getCompanionPosition(video) {
   return null;
 }
 
+/**
+ * @param {!JsonObject} playerOptions
+ * @param {string} position
+ * @param {!AmpElement} apesterElement
+ * @param {!JsonObject} consentObj
+ */
+function addCompanionAvElement(
+  playerOptions,
+  position,
+  apesterElement,
+  consentObj
+) {
+  const size = getCompanionVideoAdSize(apesterElement);
+  const refreshInterval = 30;
+  const ampAvAd = createElementWithAttributes(
+    /** @type {!Document} */ (apesterElement.ownerDocument),
+    'amp-ad',
```

Comment:
@ampproject/wg-monetization Is it standard for extensions to create `amp-ad` elements?

Reply:
I have not seen many doing this, but we are not against it if it does not run afoul of any AMP fundamentals (no layout shift, etc). However note that if the page does not have `amp-ad.js` script tag, this won't work.

Reply:
@alanorozco Hi, thanks for the question. All logic for video ad block was implemented before earlier, no new elements adding. This PR just add new video provider. If user setting up ad campaign ad block height of the ad block will be reserved above or below Apester unit (media). There won't be CLS issue there. I made small video how it looks

https://user-images.githubusercontent.com/1662080/136165781-a87b6cb5-5450-49fa-ae9d-304130375a23.mov

Reply:
@alanorozco it's a test page from local env without text, to make screen I added random text to the page:

https://user-images.githubusercontent.com/1662080/136375657-52e271fc-b819-4f87-91f1-2a7b22ac27c6.mov

Reply:
@alanorozco Also, I put video screen of Apester unit without amp-ad:

Uploading Screen Recording 2021-10-07 at 2.36.30 PM.mov…

Reply:
@alanorozco @powerivq Hi, do you need any additional information from our side?

Reply:
@golovatuy That's clearly layout shifted. The issue is that the element is being added *outside* `<amp-apester-media>`. The only way to fix this is to ensure that you don't change height of content, or insert content in between.

Reply:
@alanorozco I think I found solution, If we reserve some extra height for "amp-apester-media" element for ad block (using for example some additional attribute like extraHeightTop="350").
`<amp-apester-media height="450" extraHeightTop="350">`
Common height of the amp-apester-media and amp-ad will be in this example 800 px without any shifting.
What do you think?

Reply:
I think it can go like this:

The height of amp-apester-media should include the space for ads. During the layoutCallback, you insert `amp-ad` within amp-apeser-media element, before the video player.

This way, LCS is guaranteed not to occur.